assignment3/solve.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding: UTF-8 -*-
+
+import unittest
+
+
+class BorNode(object):
+    def __init__(self):
+        self.edges = {}
+        self.word_end = False
+
+
+class Bor(object):
+    """
+    For solving this problem I used Bor data structure.
```

Comment:
Actually this is prefix tree (or trie).
You can find explanation [here](https://en.wikipedia.org/wiki/Trie).
I named this class `Bor` because in russian this data structure has name bor like Бор : )